code/game/objects/items/implants/implant_misc.dm

```diff
@@ -109,3 +109,43 @@
 	name = "implanter (internal syndicate radio)"
 	imp_type = /obj/item/implant/radio/syndicate
 
+/obj/item/implant/adrenalin
+	name = "adrenal implant"
+	desc = "Removes all stuns."
+	icon_state = "adrenal"
+	uses = 3
+
+/obj/item/implant/adrenalin/get_data()
+	var/dat = {"<b>Implant Specifications:</b><BR>
+				<b>Name:</b> Cybersun Industries Adrenaline Implant<BR>
+				<b>Life:</b> Five days.<BR>
+				<b>Important Notes:</b> <font color='red'>Illegal</font><BR>
+				<HR>
+				<b>Implant Details:</b> Subjects injected with implant can activate an injection of medical cocktails.<BR>
+				<b>Function:</b> Removes stuns, increases speed, and has a mild healing effect.<BR>
+				<b>Integrity:</b> Implant can only be used three times before reserves are depleted."}
+	return dat
+
+/obj/item/implant/adrenalin/activate()
+	. = ..()
+	uses--
+	to_chat(imp_in, "<span class='notice'>You feel a sudden surge of energy!</span>")
+	imp_in.SetStun(0)
+	imp_in.SetKnockdown(0)
+	imp_in.SetUnconscious(0)
+	imp_in.SetParalyzed(0)
+	imp_in.SetImmobilized(0)
+	imp_in.adjustStaminaLoss(-75)
+	imp_in.SetKnockdown(0)
+	imp_in.set_resting(FALSE)
+	imp_in.update_mobility()
+
+	imp_in.reagents.add_reagent(/datum/reagent/medicine/synaptizine, 10)
+	imp_in.reagents.add_reagent(/datum/reagent/medicine/omnizine, 10)
+	imp_in.reagents.add_reagent(/datum/reagent/medicine/stimulants, 10)
```

Comment:
These should be lowered. This is 200 seconds of omnizine healing, 100 seconds of speedy gonzales. Omnizine no higher than 1.5u, stimulants no higher than 2u.

Reply:
Actually no, this is stupid. Who will buy a 3 use implant with 2u of stimulants and 1.5u of omnizine for 8 TC if you can just buy a stimpack that costs lower(5 TC) and contains 50u of stimulants. So having 10 of each reagents is okay.

Reply:
You pay for the ability to use it after getting stunned, as well as the instant effects that the stimpack does not have.
In fact, the omnizine doesn't fit in this at all, and should be removed. Its a remnant from the times before cobbychems.

Reply:
I think it's to deal with the synaptizine toxin.